cmd/thanos/query.go

```diff
@@ -28,18 +31,23 @@ import (
 	"github.com/prometheus/prometheus/promql"
 	"github.com/prometheus/tsdb/labels"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
 // registerQuery registers a query command.
 func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string) {
 	cmd := app.Command(name, "query node exposing PromQL enabled Query API with data retrieved from multiple store nodes")
 
-	grpcBindAddr, httpBindAddr, newPeerFn := regCommonServerFlags(cmd)
+	grpcBindAddr, httpBindAddr, insecure, srvcert, srvkey, srvclientca, newPeerFn := regCommonServerFlags(cmd)
 
 	httpAdvertiseAddr := cmd.Flag("http-advertise-address", "Explicit (external) host:port address to advertise for HTTP QueryAPI in gossip cluster. If empty, 'http-address' will be used.").
 		String()
 
+	cert := cmd.Flag("grpc-client-tls-cert", "TLS Certificates to use to identify this client to the server").Default("tls.crt").String()
```

Comment:
What about insecure option?

Reply:
yes! Let's not mix server options for client calls.. (:

Reply:
Okay, I've split out the client and server insecure options.
Must stress, haven't testing this yet (or even run the unit tests, dep is failing to finish for me at the moment, I'm actually building via Go 1.11 modules). I'll sort that ASAP.
I'm hoping to get around to testing this in the next day or so (I'm rewriting our prometheus config tooling at the moment which is related to this post).